src/java/org/apache/cassandra/repair/autorepair/RepairTokenRangeSplitter.java

```diff
@@ -0,0 +1,1001 @@
+                    FileUtils.stringifyFileSize(approximateBytesPerSplit), approximatePartitionsPerSplit
+        );
+        return splits;
+    }
+
+    private Collection<Range<Token>> getTokenRanges(boolean primaryRangeOnly, String keyspaceName)
+    {
+        // Collect all applicable token ranges
+        Collection<Range<Token>> wrappedRanges;
+        if (primaryRangeOnly)
+        {
+            wrappedRanges = TokenRingUtils.getPrimaryRangesForEndpoint(keyspaceName, FBUtilities.getBroadcastAddressAndPort());
+        }
+        else
+        {
+            wrappedRanges = StorageService.instance.getLocalRanges(keyspaceName);
+        }
+
+        // Unwrap each range as we need to account for ranges that overlap the ring
+        List<Range<Token>> ranges = new ArrayList<>();
+        for (Range<Token> wrappedRange : wrappedRanges)
+        {
+            ranges.addAll(wrappedRange.unwrap());
+        }
+        return ranges;
+    }
+
+    private List<SizeEstimate> getRangeSizeEstimate(String keyspace, String table, Collection<Range<Token>> tokenRanges)
+    {
+        List<SizeEstimate> sizeEstimates = new ArrayList<>();
+        for (Range<Token> tokenRange : tokenRanges)
+        {
+            logger.debug("Calculating size estimate for {}.{} for range {}", keyspace, table, tokenRange);
+            try (Refs<SSTableReader> refs = getSSTableReaderRefs(repairType, keyspace, table, tokenRange))
+            {
+                SizeEstimate estimate = getSizesForRangeOfSSTables(repairType, keyspace, table, tokenRange, refs);
+                logger.debug("Generated size estimate {}", estimate);
+                sizeEstimates.add(estimate);
+            }
+        }
+        return sizeEstimates;
+    }
+
+    @VisibleForTesting
+    static SizeEstimate getSizesForRangeOfSSTables(AutoRepairConfig.RepairType repairType, String keyspace, String table, Range<Token> tokenRange, Refs<SSTableReader> refs)
+    {
+        ICardinality cardinality = new HyperLogLogPlus(13, 25);
+        long approxBytesInRange = 0L;
+        long totalBytes = 0L;
+
+        for (SSTableReader reader : refs)
+        {
+            try
+            {
+                if (reader.openReason == SSTableReader.OpenReason.EARLY)
+                    continue;
+                CompactionMetadata metadata = (CompactionMetadata) reader.descriptor.getMetadataSerializer().deserialize(reader.descriptor, MetadataType.COMPACTION);
+                if (metadata != null)
+                    cardinality = cardinality.merge(metadata.cardinalityEstimator);
+
+                long sstableSize = reader.bytesOnDisk();
+                totalBytes += sstableSize;
+                // get the bounds of the sstable for this range using the index file but do not actually read it.
+                List<AbstractBounds<PartitionPosition>> bounds = BigTableScanner.makeBounds(reader, Collections.singleton(tokenRange));
```

Comment:
A couple issues here I realized just now.

1. `makeBounds` with a `TokenRange` was removed between 4.1 and trunk.  We also had changed visibility to public which probably shouldn't have done in hindsight.
2. `BtiTableScanner` (trie-based index) is now a thing, and we need to account for that.
3. We probably don't need to use `makeBounds` directly;  I believe the only two things it was use for was checking whether the SSTable encompassed the range  (which it should since we pre-filter), and to get the approximate bytes in the range.  In fact it looks like there's an API method on `SSTable` called `onDiskSizeForPartitionPositions` that will do this for us in an implementation agnostic way.